src/altinn-app-frontend/src/shared/components/ReadyForPrint.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * This element only serves to let our PDF generator know the app is ready and have rendered its content.
+ * It should be included in the app DOM for every possible execution path, except those where we're showing
+ * loading indicators to the user while waiting for content to get ready.
+ */
+export function ReadyForPrint() {
+  const [imagesLoaded, setImagesLoaded] = React.useState(false);
+
+  React.useLayoutEffect(() => {
+    const promises = [];
+
+    const imageLoadPromise = (img: HTMLImageElement) => {
+      return new Promise((res) => {
+        img.addEventListener('load', res);
+        img.addEventListener('error', res);
+      });
+    };
+
+    document.querySelectorAll('img').forEach((image) => {
+      image.complete || promises.push(imageLoadPromise(image));
+    });
+
+    Promise.all(promises).then(() => {
```

Comment:
Can we be sure that after all images are loaded/painted, all other content has been loaded? 

What about pages where there are no images?

Reply:
No, we can't really be sure of that - but I can't think of anything else that is added to the DOM but won't be displayed until some later stage (when it is loaded). We're not loading external CSS at any point either. The only thing I could think of is we should make sure we're not waiting for any sagas currently not fulfilled/rejected, but I can't think of a way to know that for sure (not all sagas have to result in anything either).

For pages with no images, there are no promises to await, so `Promise.all()` resolves immediately - so that shouldn't be a problem.

Reply:
> For pages with no images, there are no promises to await, so Promise.all() resolves immediately - so that shouldn't be a problem.

Wont this be a problem, since the promises resolves immediately, and the content may not have loaded yet? :) 

Any app could also potentially load external css (or js for that matter)

Reply:
By using `useLayoutEffect`, this runs after the DOM mutations, so all `img` elements should be loaded into the DOM at that point - ready for us to wait until they load.

And yes, any other resources may also be loaded - but I don't think that's a problem here. I'm assuming Puppeteer will wait for the document `readyState` to settle - but I guess I could add a state check for that as well. The core problem here is that while the page looks to be ready (i.e. the `readyState` indicates the page is ready, and the DOM has finished painting), the Altinn app is still not ready (because our initializing sagas are still running, fetching data, and the page is showing you the loading indicator).